tests: include: Remove confusing string tc_start()

punitvara opened the pull request:
This patch removes confusing strnig tc_start() and
unnecessary print line from output of all test cases which use
ztest framework.

Signed-off-by: Punit Vara <punit.vara@intel.com>

nashif commented on tests/include/tc_util.h:
```diff
@@ -67,22 +67,19 @@
 
 #define TC_PRINT(fmt, ...) PRINT_DATA(fmt, ##__VA_ARGS__)
-#define TC_START(name) PRINT_DATA("tc_start() - %s\n", name)
+#define TC_START(name) PRINT_DATA("start() - %s\n", name)
```
come to think about this, this will be also confusing, so what about just "starting test <testname>"?

punitvara replied:
Ok

punitvara replied:
I was trying to keep one word but I updated with "starting test"

nashif commented on tests/include/tc_util.h:
```diff
@@ -67,22 +67,19 @@
 
 #define TC_END_REPORT(result)                               \
 	do {                                                    \
-		PRINT_LINE;                                         \
```
looking at the output with this change and what we had before, maybe the solution was to fix how we use the lines instead of just removing them, we had:
```
***** BOOTING ZEPHYR OS v1.8.99 - BUILD: Aug 15 2017 19:16:37 *****
Running test suite test_sema_api
tc_start() - test_sema_thread2thread
===================================================================
PASS - test_sema_thread2thread.
tc_start() - test_sema_thread2isr
===================================================================
PASS - test_sema_thread2isr.
tc_start() - test_sema_reset
===================================================================
PASS - test_sema_reset.
tc_start() - test_sema_count_get
===================================================================
PASS - test_sema_count_get.
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

note how the lines were in the wrong position? it should be:

```
***** BOOTING ZEPHYR OS v1.8.99 - BUILD: Aug 15 2017 19:16:37 *****
Running test suite test_sema_api
===================================================================
tc_start() - test_sema_thread2thread
PASS - test_sema_thread2thread.
===================================================================
tc_start() - test_sema_thread2isr
PASS - test_sema_thread2isr.
===================================================================
tc_start() - test_sema_reset
PASS - test_sema_reset.
===================================================================
tc_start() - test_sema_count_get
PASS - test_sema_count_get.
===================================================================
PROJECT EXECUTION SUCCESSFUL
```
That already looks much better, and if logging is enabled, the log messages would be enclosed within the same lines of the same test.

punitvara replied:
@nashif  Done. We will get === lines two times with submitted changes.

nashif commented on tests/ztest/include/ztest_log.h:
```diff
@@ -0,0 +1,22 @@
+#ifndef __ZTEST_LOG_H__
+#define __ZTEST_LOG_H__
+
+#if CONFIG_ZTEST_LOGGING
```
we should just use SYS_LOG macros instead of introducing another set of logging infrastructure.

punitvara replied:
Okay. I was thinking the same but I though give it a shot for ztest logs.

punitvara committed:
tests: ztest: remove confusing tc_start() in ztest framework

This patch removes "tc_start()" string with "starting test" and
arranges PRINT_LINE appropriatly.

Signed-off-by: Punit Vara <punit.vara@intel.com>
